streams/cluster/src/test/java/com/hortonworks/streamline/streams/cluster/register/impl/HBaseServiceRegistrarTest.java

```diff
@@ -53,6 +56,11 @@ public void testRegister() throws Exception {
             ManualServiceRegistrar.ConfigFileInfo hiveSiteXml = new ManualServiceRegistrar.ConfigFileInfo(HBASE_SITE_XML, is);
             registrar.register(cluster, new Config(), Lists.newArrayList(hiveSiteXml));
         }
+
+        Service hbaseService = environmentService.getServiceByName(cluster.getId(), HBaseServiceRegistrar.SERVICE_NAME_HBASE);
+        assertNotNull(hbaseService);
```

Comment:
Yes I think you need to file an issue, and before that, need to make consensus among team. Personally I'm OK to make it Optional but not sure to throw specific exception.